### ISSUE TITLE
Ugly TaskCanceledException when gracefully shutting down container

### ISSUE THREAD

**Issue by gitfool** [wintoncode/Winton.Extensions.Configuration.Consul#91]
I'm wiring up consul configuration using:
```c#
...
var address = new UriBuilder(config.Scheme, config.Host, config.Port).Uri;
builder.AddConsul("mps-config-environment/yaml",
    options =>
    {
        options.ConsulConfigurationOptions = client => client.Address = address;
        options.Parser = new YamlConfigurationParser();
        options.Optional = true;
        options.ReloadOnChange = true;
    });
```
Then interactively running a dotnet core 3.1 container and shutting it down via ctrl+c:
```
docker run -it -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY -e AWS_REGION -e ASPNETCORE_ENVIRONMENT=Development -P devops/chaos-monkey
[22:47:08 INF] (Lifetime) Now listening on: http://[::]:80
[22:47:08 INF] (AWSSDK) Found credentials using the AWS SDK's default credential search
[22:47:10 INF] (Lifetime) Application started. Press Ctrl+C to shut down.
[22:47:10 INF] (Lifetime) Hosting environment: Development
[22:47:10 INF] (Lifetime) Content root path: /app
^C[22:47:33 INF] (Lifetime) Application is shutting down...
chaos-monkey exception
System.AggregateException: One or more errors occurred. (A task was canceled.)
 ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout)
   at Winton.Extensions.Configuration.Consul.ConsulConfigurationProvider.Dispose()
   at Microsoft.Extensions.Configuration.ConfigurationRoot.Dispose()
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.DisposeAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.Extensions.Hosting.Internal.Host.DisposeAsync()
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(IHost host)
   at Pharos.DevOps.ChaosMonkey.Program.<>c.<Main>b__0_0(String[] args) in D:\Devel\Mps\devops-chaos-monkey\Service\Program.cs:line 19
   at Pharos.Extensions.Logging.HostRunner.Run(String appName, String[] args, Action`1 runner)
```
`TaskCanceledException` thrown from [ConsulConfigurationProvider.Dispose](https://github.com/wintoncode/Winton.Extensions.Configuration.Consul/blob/master/src/Winton.Extensions.Configuration.Consul/ConsulConfigurationProvider.cs#L48).

Perhaps the cancellation token should be passed to `Task.Wait` as well as being wrapped in a try-catch to swallow any cancellation related exception?

**Comment by Choc13** [comment]
Thanks for raising this. I wonder if there is a more graceful way to shut this down without completely suppressing all error messages? I'll take a look into it.

**Comment by jamarino** [comment]
I ran into the same issue today while updating NuGet packages in a project.

I've submitted a pull request that suppresses the `TaskCanceledException` where it is thrown in the PollingLoop by adding `.ContinueWith()`:
```
await Task.Delay(wait, _cancellationTokenSource.Token)
    .ContinueWith(_ => { }); // suppress cancellation exceptions
```
This should not affect any other error handling.

**Comment by Choc13** [comment]
This has been fixed in 3.0.1.

**Comment by gitfool** [comment]
@Choc13 should this be available in NuGet by now, or is that a separate release process?

**Comment by Choc13** [comment]
There seems to be an issue with the NuGet API key that I’ve not had time to figure out yet. I’ll try and get it fixed this evening.

**Comment by Choc13** [comment]
Fixed the key and 3.0.1 is now published.
